src/js/settings/components/provider-settings/together-ai-images.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * WordPress dependencies
+ */
+import { SelectControl, ToggleControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { SettingsRow } from '../settings-row';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { STORE_NAME } from '../../data/store';
+import { TogetherAISettings } from './together-ai';
+
+/**
+ * Component for the Together AI Images Provider settings.
+ *
+ * @param {Object}  props              Component props.
+ * @param {boolean} props.isConfigured Whether the provider is configured.
+ *
+ * @return {React.ReactElement} TogetherAIImagesSettings component.
+ */
+export const TogetherAIImagesSettings = ( { isConfigured = false } ) => {
+	const providerName = 'togetherai_image';
+	const providerSettings = useSelect(
+		( select ) =>
+			select( STORE_NAME ).getFeatureSettings( providerName ) || {}
+	);
+	const { setProviderSettings } = useDispatch( STORE_NAME );
+	const onChange = ( data ) => setProviderSettings( providerName, data );
+
+	const models = [
+		{ label: __( '-- Choose Model --', 'classifai' ), value: '' },
+	];
+
+	// Convert providerSettings.models to an array from an object.
+	if (
+		providerSettings?.models &&
+		! Array.isArray( providerSettings.models )
+	) {
+		models.push(
+			...Object.entries( providerSettings.models ).map(
+				( [ key, model ] ) => ( {
+					label: model.display_name || key,
+					value: model.id || key,
+				} )
+			)
+		);
+	}
+
+	return (
+		<>
+			{ ! isConfigured && (
+				<TogetherAISettings
+					providerSettings={ providerSettings }
+					onChange={ onChange }
+				/>
+			) }
+			<SettingsRow
+				label={ __( 'Model', 'classifai' ) }
+				description={ __(
+					'Choose the model you want to use.',
+					'classifai'
+				) }
+			>
+				<SelectControl
+					id={ `${ providerName }_model` }
+					onChange={ ( value ) => onChange( { model: value } ) }
+					value={ providerSettings?.model || '' }
+					options={ models }
+					disabled={ ! isConfigured }
```

Comment:
```suggestion
					disabled={ models.length <= 1 }
```

`isConfigured` get updated to false when we click on edit button for the configured provider. Definitely need to update variable name to something else to make it clear, but for now, in this PR would suggest to update it to something like this. So, model select dropdown get enabled once it has models to show.

Reply:
I'm fine with this change, noting my thought here was if someone wanted to change their Provider settings (in this case, the API key) a different account may have different models it supports, so having this dropdown be disabled while those settings are being modified was my thought. That said, this basically accomplishes the same thing